src/v/serde/envelope.h

```diff
@@ -60,8 +64,13 @@ struct checksum_envelope {
     static constexpr auto redpanda_serde_compat_version = CompatVersion::v;
     static constexpr auto redpanda_inherits_from_envelope = true;
     static constexpr auto redpanda_serde_build_checksum = true;
+
 };
 
+// Overhead of the envelope in bytes
+static constexpr size_t checksum_envelope_header_size
+  = envelope_header_size + 4;
```

Comment:
Because the version_t and serde_size_t types are defined separately, we don't have both in scope here to use with sizeof.  I've added comments that explain where the bytes counts come from.